packages/now-static-build/src/utils/gatsby-config.ts

```diff
@@ -0,0 +1,80 @@
+import { writeFile, copyFile, mkdir } from 'fs';
+import { join, dirname } from 'path';
+import { promisify } from 'util';
+import { isCanary } from './is-canary';
+
+const writeFilePromise = promisify(writeFile);
+const mkdirPromise = promisify(mkdir);
+const copyFilePromise = promisify(copyFile);
+
+const GATSBY_PLUGIN_NOW_NAME = 'gatsby-plugin-now';
+const GATSBY_USER_CONFIG_PATH = './__now_gatsby_config_user.js';
+
+function createGatsbyConfig(hasUserConfig: boolean) {
+  return `let userConfig = {}
+
+${hasUserConfig ? `userConfig = require('${GATSBY_USER_CONFIG_PATH}')` : ''}
+
+module.exports = {
+  ...userConfig,
+  plugins: [
+    ...(userConfig.plugins || []),
+    '${GATSBY_PLUGIN_NOW_NAME}'
+  ]
+}`;
+}
+
+export async function injectGatsbyConfig(entrypointDir: string) {
+  if (!isCanary()) {
+    return;
+  }
+
+  try {
+    // first, we copy gatsby-plugin-now to plugins
+    const gatsbyPluginNowPath = join(
+      entrypointDir,
+      'plugins',
+      GATSBY_PLUGIN_NOW_NAME
+    );
+
+    try {
+      await mkdirPromise(dirname(gatsbyPluginNowPath));
+    } catch (err) {
+      // plugins folder already exists, ignore error
+    }
+
+    await mkdirPromise(gatsbyPluginNowPath);
+
+    await Promise.all(
+      ['gatsby-node.js', 'package.json', 'index.js'].map(file =>
+        copyFilePromise(
+          require.resolve(`./gatsby-plugin-now/${file}`),
+          join(gatsbyPluginNowPath, file)
```

Comment:
How would `./gatsby-plugin-now/index.js` exist at runtime? This seems like it would only work because we have a monorepo that will resolve properly but it will fail for real deployments.

Reply:
It is copied to dist at build time. See `build.sh`.